docs/dynamic-programming/unbounded-knapsack.mdx

```diff
@@ -0,0 +1,207 @@
+---
+title: Unbounded knapsack
+slug: /dynamic-programming/unbounded-knapsack
+---
+
+## Introduction
+
+Given a knapsack of weight W and n number of items, we need to calculate the maximum value of the items that can be packed into the knapsack such that we can choose any item any number of times. Here we assume that each item has an infinite supply.
+
+## Explanation
+
+ We use dynamic programming to solve this problem. We start from the nth item. There are two possibilities at each step: the weight of the item is less than the knapsack W or greater than the weight of the knapsack W. If the weight of the item is greater than the weight of the knapsack W, then we cannot include the item in it. But if the weight is less than the weight of the knapsack, then we again have two choices to maximize the value of the knapsack: we either choose it or not. If we donot choose it, then the next time we select from n-1 items available to us. If we choose the nth item, then next time we can choose from n items again. So there is a problem of smaller sub problems and so dynamic programming can be used to solve the problem.
+
```

Comment:
Adding diagrams would be preferred here

Reply:
I think it is difficult to explain through diagram in this case.